src/AxesUtil/AxesUtil.ts

```diff
@@ -10,13 +9,46 @@ import {
   timeWeek,
   timeYear
 } from 'd3-time';
-import {axisBottom, axisLeft} from 'd3-axis';
-import formatLocale from 'd3-format/src/locale.js';
-import select from 'd3-selection/src/select';
-import deDE from 'd3-time-format/locale/de-DE.json';
-import enUS from 'd3-time-format/locale/en-US.json';
-import FormatdeDE from 'd3-format/locale/de-DE.json';
-import FormatenUS from 'd3-format/locale/en-US.json';
+import { axisBottom, axisLeft, Axis, AxisDomain, AxisScale } from 'd3-axis';
+import { ScaleLinear } from 'd3-scale';
+import { FormatLocaleDefinition, formatLocale } from 'd3-format';
+import { select } from 'd3-selection';
+import { TimeLocaleDefinition } from 'd3-time-format';
+import { Moment } from 'moment';
+import moment = require('moment');
+import { Scale } from 'src/ScaleUtil/ScaleUtil';
+import { NodeSelection } from 'src/BaseUtil/BaseUtil';
+const deDE: TimeLocaleDefinition = require('d3-time-format/locale/de-DE.json');
+const enUS: TimeLocaleDefinition = require('d3-time-format/locale/en-US.json');
+const FormatdeDE: FormatLocaleDefinition = require('d3-format/locale/de-DE.json');
+const FormatenUS: FormatLocaleDefinition = require('d3-format/locale/en-US.json');
+
+export interface AxisConfiguration {
+  labelRotation?: number;
+  display?: boolean;
+  labelSize?: number;
+  labelPadding?: number;
+  labelColor?: string;
+  label?: string;
+  ticks?: number;
+  tickSize?: number;
+  tickPadding?: number;
+  tickValues?: number[];
+  locale?: 'de' | 'en';
+  autoTicks?: boolean;
+  scale?: string;
+  format?: string;
+  min?: any;
+  max?: any;
+  harmonize?: boolean;
+  factor?: number;
+  orientation: 'x' | 'y';
+  gridWidth?: string;
+  gridColor?: string;
+  showGrid?: boolean;
+  gridOpacity?: number;
+  sanitizeLabels?: boolean;
+}
```

Comment:
Are there any required Options?

Reply:
Not really. The code was written to cope even with empty configuration objects in mind, that's why they're all optional.